src/components/extensive/Carousel/Carousel.tsx

```diff
@@ -45,9 +47,15 @@ const Carousel = <T extends Record<any, any>>({
 
   const swiperButtonSize = smallSwiperButtons ? 12 : 24;
 
+  const handleSlideChange = (swiper: { activeIndex: any }) => {
+    if (setSelectecImage) {
+      setSelectecImage(swiper.activeIndex);
+    }
```

Comment:
The callback is misspelled, should be `setSelectedImage`. Also, this can be simplified to
```
const handleSlideChange = (swiper: { activeIndex: any }) => setSelectedImage?.(swiper.activeIndex);
```

Reply:
Thank you for catching that typo. I've corrected the spelling to setSelectedImage and simplified the callback as suggested